json-path/src/test/java/com/jayway/jsonpath/OptionsTest.java

```diff
@@ -40,6 +85,31 @@ public void a_definite_path_is_not_returned_as_list_by_default() {
         assertThat(using(conf).parse("{\"foo\" : \"bar\"}").read("$.foo")).isInstanceOf(String.class);
     }
 
+    /**
+     * Test to check if a map instance is returned, without ALWAYS_RETURN_LIST set.
+     * 
+     */
+    @Test
+    public void a_definite_path_is_not_returned_as_list_by_default_map_readroot() {
+
+        Configuration conf = Configuration.defaultConfiguration();
+        Object root = using(conf).parse("{\"foo\" : \"bar\"}").readRoot(new String[] {"$.foo"});
+        assertThat(root).isInstanceOf(Map.class);
+    }
+
+    /**
+     * Test to check if a map instance is returned, even with ALWAYS_RETURN_LIST set.
+     * 
+     */
+    @Test
+    public void a_definite_path_is_returned_as_list_by_default_map_readroot() {
+
+        Configuration conf = Configuration.defaultConfiguration();
+        conf.setOptions(Option.ALWAYS_RETURN_LIST);
```

Comment:
Change this, in another PR, to illegalArgException . we don't support always_return_list + readRoot(). does not make sense as our default is always return Map